### DISCUSSION TITLE
ERROR: must be owner of table in pgAdmin4

### DISCUSSION THREAD

**Discussion by binajmen** [supabase/supabase#4825] [Questions]
Hi everyone,

Is there a way to allow changes within pgAdmin4 without having to change the owner of the table(s) in the Supabase interface like suggested in https://github.com/supabase/supabase/discussions/3271?

Thank you.

**Comment by awalias** [comment]
you can grant certain permissions to the postgres user depending on what you're trying to do inside pgAdmin e.g.

`GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA public TO postgres;`

some more examples here: https://tableplus.com/blog/2018/04/postgresql-how-to-grant-access-to-users.html

>allow changes

depending on whether you just want to edit data or actually alter the table structure (alter columns etc.)

**Reply by binajmen** [reply]
@awalias Thank you, sounds nice! However, is there an impact on security regarding the anon-key which is public when used in a web app?

**Reply by asciidiego** [reply]
@awalias does this mean that, in the Supabase-hosted PostgreSQL instance, the user `postgres` does NOT really have the permissions to, say, alter the structure of a table? And then, when we alter the structure of a table using the Supabase UI what is going on is that the PG connection (and hence the query) is being done using the `supabase_admin` role?

**Comment by bohanwood** [comment]
Solution: `GRANT supabase_admin TO postgres;`

**Reply by ozozozd** [reply]
@bohanyang a much superior solution than [the one in the docs.](https://supabase.com/docs/guides/resources/supabase-cli/managing-environments#permission-denied-on-db-remote-commit)

**Reply by pcardosolei** [reply]
it is a superior solution but has a more other side effects. you are giving all permisisons to a account that shouldnt be a superadmin

**Reply by sonnyvesali** [reply]
What ended up helping me was assigning the owner on creation, in my case I created a new schema from the supabase dashboard so what was previously `create schema if not exists "internal_functions";` was fixed by `create schema if not exists "internal_functions" AUTHORIZATION postgres;`
